.github/workflows/cypress-workflow.yml

```diff
@@ -0,0 +1,129 @@
+name: E2E Cypress tests
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+jobs:
+  tests:
+    name: Run Cypress E2E tests
+    runs-on: ubuntu-latest
+    env:
+      # prevents extra Cypress installation progress messages
+      CI: 1
+      # avoid warnings like "tput: No value for $TERM and no -T specified"
+      TERM: xterm
+    steps:
+      - name: Set up JDK
+        uses: actions/setup-java@v1
+        with:
+          # TODO: Parse this from security plugin
+          java-version: 13
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          path: security
+          repository: opendistro-for-elasticsearch/security
+      - name: Checkout Security Kibana plugin
+        uses: actions/checkout@v2
+        with:
+          path: security-kibana-plugin
+      - name: Get Kibana version
+        id: kibana_version
+        run: |
+          echo "::set-output name=kibana_version::$(node -p "(require('./security-kibana-plugin/package.json').kibana.version).match(/[.0-9]+/)[0]")"
+      - name: Get Opendistro version
+        id: opendistro_version
+        run: |
+          echo "::set-output name=opendistro_version::$(node -p "(require('./security-kibana-plugin/package.json').version).match(/[.0-9]+/)[0]")"
+      - name: Get Cypress version
+        id: cypress_version
+        run: |
+          echo "::set-output name=cypress_version::$(node -p "(require('./security-kibana-plugin/package.json').devDependencies.cypress).match(/[.0-9]+/)[0]")"
+      - name: Run elasticsearch with plugin
+        run: |
+          wget https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-${{ steps.kibana_version.outputs.kibana_version }}-darwin-x86_64.tar.gz
+          tar -xzf elasticsearch-oss-${{ steps.kibana_version.outputs.kibana_version }}-darwin-x86_64.tar.gz
+          cd elasticsearch-${{ steps.kibana_version.outputs.kibana_version }}/
+          bin/elasticsearch-plugin install -b https://d3g5vo6xdbdb9a.cloudfront.net/downloads/elasticsearch-plugins/opendistro-security/opendistro_security-${{ steps.opendistro_version.outputs.opendistro_version }}.zip
+          bash plugins/opendistro_security/tools/install_demo_configuration.sh -y -i
+          bin/elasticsearch &
+          sleep 60
+          curl https://localhost:9200 -u admin:admin -k
+      - name: Checkout Kibana
+        uses: actions/checkout@v2
+        with:
+          repository: opendistro-for-elasticsearch/kibana-oss
+          ref: ${{ steps.kibana_version.outputs.kibana_version }}
+          token: ${{ secrets.KIBANA_OSS_TOKEN }}
+          path: kibana
+      - name: Get node and yarn versions
+        id: versions
+        run: |
+          echo "::set-output name=node_version::$(node -p "(require('./kibana/package.json').engines.node).match(/[.0-9]+/)[0]")"
```

Comment:
kibana has a `.node-version` file to note the nodeJS version

Reply:
Does .node-version file have higher priority than that in package.json? If yes, I can switch to it.

Reply:
kibana determines the local node version matches node version in `.node-version` file, and failed the build if mismatch. and using .node-version, you can avoid parsing node version from json file

Reply:
Thanks. Switched to read from the .node-version file directly